src/qt/bitcoin.cpp

```diff
@@ -459,8 +459,11 @@ int GuiMain(int argc, char* argv[])
     SetupUIArgs();
     std::string error;
     if (!node->parseParameters(argc, argv, error)) {
+        InitError(strprintf("Error parsing command line arguments: %s\n", error));
```

Comment:
Just a suggestion, but it would make sense to change `InitError()` to `node->initError()` here, adding a new `Node::initError()` method like existing [`Chain::initError()`](https://github.com/bitcoin/bitcoin/blob/4fcccdac785e09ad5627b3bf4811dfba353693e8/src/interfaces/chain.h#L218-L219) and [`ChainImpl::initError()`](https://github.com/bitcoin/bitcoin/blob/4fcccdac785e09ad5627b3bf4811dfba353693e8/src/interfaces/chain.cpp#L343) methods.

This would prevent error logs being dropped with #10102. Calling `InitError` in the `bitcoin-gui` process there can't have any effect because `InitError` uses global variables initialized in the `bitcoin-node` process. (Ideally the `InitError` function wouldn't even be accessible to gui code, but there is no enforcement yet for GUI code not accessing node globals currently, unlike the enforcement for wallet code not accessing node globals that was added in 78a2fb55c97fbc26f7b74c5b1fb999a2aff8ce88.)

Feel free to ignore this suggestion if it is confusing or too much work, since I can clean this up later.

Reply:
Excellent suggestion, but I am not sure how to implement this cleanly. We have the `./src/node` and `./src/interface` subdirectories, so it seems I'd have to split the global `uiInterface`+`InitError` out from `ui_interface` into `./src/node/ui` and move the remainder of `ui_interface` into `./src/interface/ui`?

I am happy to cherry-pick something if you code this up.

Reply:
76b1c671f97a97e21326715c8ea21f5584ab5111 (https://github.com/ryanofsky/bitcoin/commits/pr/initerror) is the suggestion. I might not have described it clearly before.

Reply:
Thx. Cherry-picked into this branch.